File: 2021/06/06/main.go
package main

import "sort"

func longestConsecutive(nums []int) int {
	if len(nums) <= 1 {
		return len(nums)
	}
	sort.Ints(nums)
	mx, r := 0, 1
	for i := 1; i < len(nums); i++ {
		if nums[i-1] == nums[i]-1 {
			r++
		} else if nums[i] != nums[i-1] {
			r = 1
		}
		if r > mx {
			mx = r
		}
	}
	return mx
}

File: 2021/06/06/main.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) <= 1:
            return len(nums)
        nums.sort()
        mx, r = 0, 1
        for i in range(1, len(nums)):
            if nums[i-1] == nums[i] - 1:
                r += 1
            elif nums[i] != nums[i-1]:
                r = 1
            mx = max(mx, r)
        return mx


